### DISCUSSION TITLE
Ability to change the cover page of manga

### DISCUSSION THREAD

**Discussion by dragow0lf** [xgi/houdoku#118] [Feature Requests]
Another feature request, which lets us change the cover page of stuff in our library
Would be really useful, And I think it should be possible cause Local Manga already has that feature
But then again, Don't know if it would really be possible cause the covers for manga are taken from the actual site the manga is hosted in, and It's possible to change the cover in local cause we are basically the ones hosting them

**Comment by xgi** [comment]
The reason this isn't currently supported is because of how the metadata for a series is stored. It looks something like this:
```
{
  "title": "Chainsaw Man",
  "description": "Broke young man + chainsaw dog demon = Chainsaw Man!",
  "authors": [ "Fujimoto Tatsuki" ],
  "genres": [ "ACTION", "COMEDY","HORROR" ],
  "demographic": "SHOUNEN",
  "status": "Completed",
  "originalLanguageKey": "JAPANESE",
  "remoteCoverUrl": "https://uploads.mangadex.org/covers/a77742b1-befd-49a4-bff5-1ad4e6b0ef7b/3713217f-e35c-44d9-b569-17ba91eba4b2.jpg.512.jpg",
  ... bunch of other stuff
}
```

The reason I only currently support editing for local series is that all of these fields can (and usually should) get updated when the data from the extension changes -- including the cover. If I made the fields editable then they would get replaced the next time the extension reloaded. I could add a flag saying "this series was manually edited, don't pull changes from the extension" but then if you changed just one field you wouldn't get updates made to the others (like status, description, etc.).

I could instead let users "override" certain fields (like the cover), but from a UX perspective it makes it a bit difficult to show when something is overridden (and there'd need to be specific handling for "reverting" to the data from the extension).

This isn't impossible to work out, I just don't have a clear picture of how I'd do it yet.
